contexts/arbo-context.tsx

```diff
@@ -47,51 +47,44 @@ function filterData(data: any[], filters: { [key: string]: string[] }): any[] {
     return filterKeys.every((key: string) => {
       if (!filters[key].length) return true;
 
-      if(key === "end_date") {
-        const filterEndDate = parseISO(filters["end_date"][0]);
+      if (key === "end_date") {
+        const filterEndDate = new Date(filters["end_date"][0]);
+        const filterStartDate = new Date(filters["start_date"][0]);
 
-        if(!filterEndDate) {
-          return true;
+        if (isNaN(filterEndDate.getTime())) {
+          return true; // Handle invalid date
         }
 
-        const itemDate = new Date(item.sample_end_date);
+        const itemStartDate = new Date(item.sample_start_date);
+        const itemEndDate = new Date(item.sample_end_date);
 
-        const filterEndUTC = Date.UTC(
-          filterEndDate.getUTCFullYear(),
-          filterEndDate.getUTCMonth(),
+        // Check for any overlap in the sampling period
+        return (
+          itemEndDate <= filterEndDate ||
+          (itemEndDate >= filterEndDate && itemStartDate < filterEndDate)
         );
```

Comment:
I'm curious if this is equivalent to `itemStartDate < filterEndDate`. I really like what you did with the other one with `itemEndDate >= filterStartDate` and I'm pretty sure that would work here if you also threw in your solution for the studies in the Central African Republic.